.taskcluster.yml

```diff
@@ -0,0 +1,170 @@
+      deadline: {$fromNow: '1 hour'}
+      extra:
+        github:
+          events:
+            - pull_request.opened
+            - pull_request.reopened
+            - pull_request.synchronize
+            - push
+            - release
+      payload:
+        maxRunTime: 3600
+        image: python:3.7
+        command:
+          - sh
+          - -lxce
+          - >-
+            git clone ${repository} /src &&
+            cd /src &&
+            git config advice.detachedHead false &&
+            git checkout ${head_rev} &&
+            pip install tox &&
+            tox -e py37
+      metadata:
+        name: tox py37
+        description: code linting & unit tests on py37
+        owner: ${owner}
+        source: ${repository}/raw/${head_rev}/.taskcluster.yml
+
+    - taskId: '${as_slugid("docker_build")}'
+      dependencies:
+        - '${as_slugid("py36")}'
+        - '${as_slugid("py37")}'
+      provisionerId: aws-provisioner-v1
+      workerType: releng-svc
+      created: {$fromNow: ''}
+      deadline: {$fromNow: '1 hour'}
+      extra:
+        github:
+          events:
+            - pull_request.opened
+            - pull_request.reopened
+            - pull_request.synchronize
+            - push
+            - release
+      payload:
+        capabilities:
+          privileged: true
+        maxRunTime: 3600
+        image: mozilla/taskboot:latest
+        command:
+          - taskboot
+          - --git-repository=${repository}
+          - --git-revision=${head_rev}
+          - build
+          - --write=/image.tar
+          - --tag=registry.hub.docker.com/mozillareleases/beetmoverscript:${branch_or_version}
+          - Dockerfile
+        artifacts:
+          public/beetmoverscript/image.tar:
+            expires: {$fromNow: '2 weeks'}
+            path: /image.tar
+            type: file
+      scopes:
+        - docker-worker:capability:privileged
+      metadata:
+        name: docker build (${branch_or_version})
+        description: build latest docker image (${branch_or_version})
+        owner: ${owner}
+        source: ${repository}/raw/${head_rev}/.taskcluster.yml
+
+    - $if: 'branch_or_version != "pull-request"'
+      then:
+        taskId: '${as_slugid("docker_push")}'
+        dependencies:
+          - '${as_slugid("docker_build")}'
+        provisionerId: aws-provisioner-v1
+        workerType: releng-svc
+        created: {$fromNow: ''}
+        deadline: {$fromNow: '1 hour'}
+        extra:
+          github:
+            events:
+              - push
+              - release
+        payload:
+          features:
+            # Needed for access to secret
+            taskclusterProxy: true
+          maxRunTime: 3600
+          image: mozilla/taskboot:latest
+          command:
+            - taskboot
+            - --secret=project/releng/beetmoverscript/deploy
+            - push-artifact
+        scopes:
+          - secrets:get:project/releng/beetmoverscript/deploy
```

Comment:
Yeah. It would be much better to run these things on dedicated workers that we trust. Also in the future we probably want to use taskgraph and a decision task instead of the current approach. The main idea of this patch is to start publishing docker images to dockerhub, so we can deploy to GCP. Once we have the pipeline set up we can replace the underlying approach easily.